Add `<(color|hue)-interpolation-method>`; de-duplicate `<hue>` descriptions (#26775)
mdn/content: files/en-us/web/css/color-interpolation-method/index.md

---
title: <color-interpolation-method>
slug: Web/CSS/color-interpolation-method
page-type: css-type
browser-compat:
  - css.types.color.color-mix
  - css.types.image.gradient.conic-gradient.interpolation_color_space
  - css.types.image.gradient.linear-gradient.interpolation_color_space
  - css.types.image.gradient.radial-gradient.interpolation_color_space
  - css.types.image.gradient.repeating-conic-gradient.interpolation_color_space
  - css.types.image.gradient.repeating-linear-gradient.interpolation_color_space
  - css.types.image.gradient.repeating-radial-gradient.interpolation_color_space
spec-urls: https://drafts.csswg.org/css-color/#interpolation-space
---

{{CSSRef}}

The **`<color-interpolation-method>`** [CSS](/en-US/docs/Web/CSS) [data type](/en-US/docs/Web/CSS/CSS_Types) represents the color space used for interpolation between {{CSSXref("&lt;color&gt;")}} values. It can be used to override the default interpolation color space for color-related functional notations such as {{CSSXref("color_value/color-mix", "color-mix()")}} and {{CSSXref("gradient/linear-gradient", "linear-gradient()")}}.

> When mixing and [combining](https://drafts.csswg.org/css-values/#combining-values) colors, the interpolation color space defaults to Oklab.

Author: I plan to mention this in the "Interpolation" section of `<color>` as well. However, the pros and cons of this description on the `<color-interpolation-method>` page are:

  - **Con 1:** This is more about how `<color>` values interpolate.
  - **Con 2:** It may incorrectly suggest that `<color-interpolation-method>` defaults to `in oklab`.
  - **Pro 1:** For developers unaware of different interpolation color spaces that want to figure out why some `<color>` values don't interpolate as expected, they'll probably search for "color interpolation" and arrive at the `<color-interpolation-method>` page.

> When mixing and [combining](https://drafts.csswg.org/css-values/#combining-values) colors, the interpolation color space defaults to Oklab.

Author: I plan to mention this in the "Interpolation" section of `<color>` as well. However, the pros and cons of this description on the `<color-interpolation-method>` page are:

  - **Con 1:** This is more about how `<color>` values interpolate.
  - **Con 2:** It may incorrectly suggest that `<color-interpolation-method>` defaults to `in oklab`.
  - **Pro 1:** For developers unaware of different interpolation color spaces that want to figure out why some `<color>` values don't interpolate as expected, they'll probably search for "color interpolation" and arrive at the `<color-interpolation-method>` page.

---

The same arguments apply to `<hue-interpolation-method>`.

Reviewer: I think it fits here. Question: is there a difference between "mixing" and "combining"? If not, we should just use one of these to be clear.

Author: I took this wording from the [spec](https://drafts.csswg.org/css-color/#interpolation-space):

> Mixing or otherwise combining colors[…]

As I understand it, although both actions invole interpolation of colors, mixing refers to producing a single outcome color as an art term, whereas combining refers to producing a series of intermediate colors as used in CSS transitions and animations.

If this wording is confusing without context, I could simplify as below:
```suggestion
When interpolating colors, the interpolation color space defaults to Oklab.
```

```
in <rectangular-color-space>
in <polar-color-space>[ <hue-interpolation method>]
```

### Values

- `<rectangular-color-space>`
  - : One of the keywords `srgb`, `srgb-linear`, `display-p3`, `a98-rgb`, `prophoto-rgb`, `rec2020`, `lab`, `oklab`, `xyz`, `xyz-d50`, and `xyz-d65`.
- `<polar-color-space>`
  - : One of the keywords `hsl`, `hwb`, `lch`, and `oklch`.
- {{CSSXref("&lt;hue-interpolation-method&gt;")}} {{optional_inline}}
  - : The algorithm for hue interpolation. It defaults to `shorter hue`.

### Formal syntax

{{CSSSyntax}}

## Examples

### Comparing interpolation color spaces using gradients

The following example shows the effect of using different interpolation color spaces for {{CSSXref("gradient/linear-gradient", "linear-gradient()")}}.

#### HTML

```html
<div>sRGB:</div>
<div class="gradient srgb"></div>
<div>Display P3:</div>
<div class="gradient display-p3"></div>
<div>Oklab:</div>
<div class="gradient oklab"></div>
<div>Oklch (with <code>longer hue</code>):</div>
<div class="gradient oklch-longer"></div>
```

#### CSS

```css
.gradient {
  background-image: linear-gradient(
    var(--method) to right,
    #a37,
    #595

>   );

Author: Adapted from [Example 19](https://drafts.csswg.org/css-images-4/#example-9b093da0) of CSS Images 4.

>   );

Author: Adapted from [Example 19](https://drafts.csswg.org/css-images-4/#example-9b093da0) of CSS Images 4.

slug: Web/CSS/color-interpolation-method
page-type: css-type
browser-compat:
  - css.types.color.color-mix
  - css.types.image.gradient.conic-gradient.interpolation_color_space
  - css.types.image.gradient.linear-gradient.interpolation_color_space
  - css.types.image.gradient.radial-gradient.interpolation_color_space
  - css.types.image.gradient.repeating-conic-gradient.interpolation_color_space
  - css.types.image.gradient.repeating-linear-gradient.interpolation_color_space
  - css.types.image.gradient.repeating-radial-gradient.interpolation_color_space
spec-urls: https://drafts.csswg.org/css-color/#interpolation-space
---

{{CSSRef}}

The **`<color-interpolation-method>`** [CSS](/en-US/docs/Web/CSS) [data type](/en-US/docs/Web/CSS/CSS_Types) represents the color space used for interpolation between {{CSSXref("&lt;color&gt;")}} values. It can be used to override the default interpolation color space for color-related functional notations such as {{CSSXref("color_value/color-mix", "color-mix()")}} and {{CSSXref("gradient/linear-gradient", "linear-gradient()")}}.

When interpolating `<color>` values, the interpolation color space defaults to Oklab.

## Syntax

> The `<color-interpolation-method>` data type has two syntaxes:

Reviewer: ```suggestion
The `<color-interpolation-method>` specifies whether interpolation should use a rectangular color space or a polar color space with an optional hue interpolation method:
```

```

> in <rectangular-color-space>
> in <polar-color-space>[ <hue-interpolation method>]

Author: An alternative is to mimic the format on [`<easing-function>`](https://developer.mozilla.org/en-US/docs/Web/CSS/easing-function#syntax):
```suggestion
in <rectangular-color-space>;
in <polar-color-space>[ <hue-interpolation method>];
```

> in <polar-color-space>[ <hue-interpolation method>]

Reviewer: I think leaving out the semicolons is a good idea because we're not trying to provide a full selector / valid rule like:

```
background-image: linear-gradient(in oklch longer hue to right, blue, red);
```

```

### Values

- `<rectangular-color-space>`

>   - : One of the keywords `srgb`, `srgb-linear`, `display-p3`, `a98-rgb`, `prophoto-rgb`, `rec2020`, `lab`, `oklab`, `xyz`, `xyz-d50`, and `xyz-d65`.

Reviewer: nitpicking, but might clarify:

```suggestion
  - : One of the keywords `srgb`, `srgb-linear`, `display-p3`, `a98-rgb`, `prophoto-rgb`, `rec2020`, `lab`, `oklab`, `xyz`, `xyz-d50`, or `xyz-d65`.
```

- `<polar-color-space>`

>   - : One of the keywords `hsl`, `hwb`, `lch`, and `oklch`.

Reviewer: Same comment here:

```suggestion
  - : One of the keywords `hsl`, `hwb`, `lch`, or `oklch`.
```

- {{CSSXref("&lt;hue-interpolation-method&gt;")}} {{optional_inline}}

  - : The algorithm for hue interpolation. It defaults to `shorter hue`.

## Formal syntax

{{CSSSyntax}}

## Examples

### Comparing interpolation color spaces using gradients

> The following example shows the effect of using different interpolation color spaces for {{CSSXref("gradient/linear-gradient", "linear-gradient()")}}.

Reviewer: We don't have full browser compat for interpolation methods in linear gradients yet, so maybe a fallback is nice or some check with https://developer.mozilla.org/en-US/docs/Web/CSS/@supports#syntax

```css
background-image: linear-gradient(to right, blue, red); /* same as srgb, right? */

@supports (background-image: linear-gradient(in oklab, red, blue)) {
   /* we can do fancy interpolation with gradients */
}

@supports not (background-image: linear-gradient(in oklab, red, blue)) {
   /* A fallback */
}
```

> 

Reviewer: We should mention that P3 depends on the device.

#### HTML

```html
<div>sRGB:</div>
<div class="gradient srgb"></div>
<div>Display P3:</div>

> <div class="gradient display-p3"></div>

Reviewer: I think we can get rid of P3 (note above), but the direction the example is going is good.

<div>Oklab:</div>
<div class="gradient oklab"></div>
<div>Oklch (with <code>longer hue</code>):</div>
<div class="gradient oklch-longer"></div>
```

#### CSS

```css
.gradient {

>   background-image: linear-gradient(var(--method) to right, #a37, #595);

Reviewer: In general, I like the named colors for these because it's less mental overhead to understand what's being demonstrated:

```suggestion
  background-image: linear-gradient(var(--method) to right, blue, red);
```
